compiler/noirc_frontend/src/hir/def_map/mod.rs

```diff
@@ -391,6 +393,10 @@ impl TestFunction {
         self.location.file
     }
 
+    pub fn has_arguments(&self) -> bool {
+        self.has_arguments
```

Comment:
Could there be cases where this is false at one point and true at another point? E.g. if a function is modified by metaprogramming? If so, then I think we should calculate it each time instead of caching

Reply:
I don't think this is possible. This is computed after the program was compiled.